packages/core/src/CronJobManager.ts

```diff
@@ -0,0 +1,27 @@
+import { Service } from 'typedi';
+import { CronJob } from 'cron';
+import type { Workflow } from 'n8n-workflow';
+
+@Service()
+export class CronJobManager {
+	readonly cronJobs = new Map<string, CronJob[]>();
+
+	registerCron(workflow: Workflow, cronExpression: string, onTick: () => void) {
+		const cronJob = new CronJob(cronExpression, onTick, undefined, true, workflow.timezone);
+		if (this.cronJobs.has(workflow.id)) this.cronJobs.set(workflow.id, []);
```

Comment:
```suggestion
		if (!this.cronJobs.has(workflow.id)) this.cronJobs.set(workflow.id, []);
```

Reply:
yeah. already fixed this after writing the unit test  😅